modules/storage-gcs/gcs/gcs.go

```diff
@@ -0,0 +1,131 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2022 SeMI Technologies B.V. All rights reserved.
+//
+//  CONTACT: hello@semi.technology
+//
+
+package gcs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"cloud.google.com/go/storage"
+	"github.com/pkg/errors"
+	"github.com/semi-technologies/weaviate/entities/snapshots"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+const (
+	GOOGLE_APPLICATION_CREDENTIALS = "GOOGLE_APPLICATION_CREDENTIALS"
+	GOOGLE_CLOUD_PROJECT           = "GOOGLE_CLOUD_PROJECT"
+	GCLOUD_PROJECT                 = "GCLOUD_PROJECT"
+	GCP_PROJECT                    = "GCP_PROJECT"
+)
+
+type gcs struct {
+	client    *storage.Client
+	config    Config
+	projectID string
+}
+
+func New(ctx context.Context, config Config) (*gcs, error) {
+	options := []option.ClientOption{}
+	if len(os.Getenv(GOOGLE_APPLICATION_CREDENTIALS)) > 0 {
+		creds, err := google.FindDefaultCredentials(ctx)
+		if err != nil {
+			return nil, errors.Wrap(err, "find default credentials")
+		}
+		options = append(options, option.WithCredentials(creds))
+	} else {
+		options = append(options, option.WithoutAuthentication())
+	}
+	projectID := os.Getenv(GOOGLE_CLOUD_PROJECT)
+	if len(projectID) == 0 {
+		projectID = os.Getenv(GCLOUD_PROJECT)
+		if len(projectID) == 0 {
+			projectID = os.Getenv(GCP_PROJECT)
+		}
+	}
+	client, err := storage.NewClient(ctx, options...)
+	if err != nil {
+		return nil, errors.Wrap(err, "create client")
+	}
+	return &gcs{client, config, projectID}, nil
+}
+
+func (g *gcs) StoreSnapshot(ctx context.Context, snapshot snapshots.Snapshot) error {
```

Comment:
Should we pass in `snapshot` by ref now?

Reply:
I think that @aliszka is working on connecting everything together. I think that he could do this during work on his task